### ISSUE TITLE
Graph based data structure

### ISSUE THREAD

**Issue by mantono** [mantono/DuplicateSearcher#12]
Create a graph based data structure where related issues will have edges. This is in order to reduce the time complexity for lookup/similarity comparison as an array based model would have O(n²), which does not scale well for larger repositories. Due to limited time, this is a low priority issue.

**Issues it depends on:** #11

**Comment by mantono** [comment]
Current data structure/search algorithm with time complexity O(n²/2) requires 8 hours, 55 minutes and 20.015 seconds to process 15 025 issues.
